### PR TITLE
W-15838197: Remove APIs from mule exposing removed java ee classes/interfaces

### PR DESCRIPTION
**Pull request by marianorean**

### CODE REVIEW

**Comment by marianorean** [comment]
--validate

**Comment by marianorean** [comment]
--validate

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/privileged/resources/documentation/ExtensionDescriptionsSerializer.java [hunk 1]
```diff
@@ -6,91 +6,20 @@
...
-  }
 
-  public synchronized XmlExtensionDocumentation deserialize(String xml) {
-    try {
-      return (XmlExtensionDocumentation) unmarshaller.unmarshal(new ByteArrayInputStream(xml.getBytes()));
-    } catch (Exception e) {
-      throw new RuntimeException(e);
-    }
-  }
+public abstract class ExtensionDescriptionsSerializer {
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/privileged/resources/documentation/ExtensionDescriptionsSerializer.java:13]: make an interface

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/privileged/resources/documentation/XmlExtensionDocumentation.java [hunk 2]
```diff
@@ -6,85 +6,25 @@
...
-  @XmlElementWrapper(name = "sources")
-  @XmlElement(name = "source")
-  public List<XmlExtensionElementDocumentation> getSources() {
-    return sources;
-  }
-
-  public void setSources(List<XmlExtensionElementDocumentation> sources) {
-    this.sources = sources;
-  }
+public abstract class XmlExtensionDocumentation {
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/privileged/resources/documentation/XmlExtensionDocumentation.java:17]: make an interface

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/privileged/resources/documentation/XmlExtensionElementDocumentation.java [hunk 3]
```diff
@@ -6,53 +6,13 @@
...
-
-  public void setName(String name) {
-    this.name = name;
-  }
-
-  @XmlElement
-  public String getDescription() {
-    return description;
-  }
+public abstract class XmlExtensionElementDocumentation {
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/privileged/resources/documentation/XmlExtensionElementDocumentation.java:11]: make an interface

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/privileged/resources/documentation/XmlExtensionParameterDocumentation.java [hunk 4]
```diff
@@ -6,46 +6,14 @@
...
-  @Override
-  @XmlAttribute
-  public String getName() {
-    return name;
-  }
-
-  public void setName(String name) {
-    this.name = name;
-  }
+public abstract class XmlExtensionParameterDocumentation {
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/privileged/resources/documentation/XmlExtensionParameterDocumentation.java:14]: make an interface

**Review by elrodro83 — CHANGES_REQUESTED** [review]
Move remaining privileged documentation stuff to api

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/api/resources/documentation/ExtensionDescriptionsSerializer.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.api.resources.documentation;
+
+import org.mule.runtime.module.extension.internal.resources.documentation.DefaultExtensionDescriptionsSerializer;
+
+import java.io.InputStream;
+
+public interface ExtensionDescriptionsSerializer {
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/api/resources/documentation/ExtensionDescriptionsSerializer.java:13]: add javadocs

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/api/resources/documentation/XmlExtensionElementDocumentation.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.api.resources.documentation;
+
+import java.util.List;
+
+public interface XmlExtensionElementDocumentation {
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/api/resources/documentation/XmlExtensionElementDocumentation.java:11]: javadoc

File: modules/extensions-support/src/main/java/module-info.java [hunk 7]
```diff
@@ -14,9 +14,7 @@
  */
 @PrivilegedApi(
   privilegedPackages = {
-        "org.mule.runtime.module.extension.api.runtime.privileged",
-        // Used by the extensions-maven-plugin
-        "org.mule.runtime.module.extension.privileged.resources.documentation"
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/module-info.java:19]: add the export for the new package, keep the comment mentioning who is using it

**Review by elrodro83 — CHANGES_REQUESTED** [review]

**Review by elrodro83 — APPROVED** [review]

**Comment by marianorean** [comment]
--validate

**Comment by marianorean** [comment]
--validate

**Comment by marianorean** [comment]
--validate

**Pull request merged**